plugins/lua/buildingplan.lua

```diff
@@ -92,4 +94,114 @@ function construct_building_from_ui_state()
     return bld
 end
 
+--
+-- GlobalSettings dialog
+--
+
+local GlobalSettings = defclass(GlobalSettings, dialogs.MessageBox)
+GlobalSettings.focus_path = 'buildingplan_globalsettings'
+
+GlobalSettings.ATTRS{
+    settings = {}
+}
+
+function GlobalSettings:onDismiss()
+    for k,v in pairs(self.settings) do
+        -- call back into C++ to save changes
+        setSetting(k, v)
+    end
+end
+
+-- does not need the core suspended.
+function show_global_settings_dialog(settings)
+    GlobalSettings{
+        frame_title="Buildingplan Global Settings",
+        settings=settings,
+    }:show()
+end
+
+function GlobalSettings:toggle_setting(name)
+    self.settings[name] = not self.settings[name]
+end
+
+function GlobalSettings:get_setting_string(name)
+    if self.settings[name] then return 'On' end
+    return 'Off'
+end
+
+function GlobalSettings:is_setting_enabled(name)
+    return self.settings[name]
+end
+
+function GlobalSettings:make_setting_label_token(text, key, name, width)
+    return {text=text, key=key, key_sep=': ', key_pen=COLOR_GREEN,
```

Comment:
COLOR_LIGHTGREEN is usually used for keys, so I would use that (some DFHack bindings also use COLOR_LIGHTRED, but that's less necessary when the entire screen is implemented by DFHack)

Reply:
Ack. Thanks for fixing!